### ISSUE TITLE
[Feature] Add bucket sorting (order)

### ISSUE THREAD

**Issue by leni8ec** [ScoopInstaller/Scoop#5131]
Is it possible to add the use of manual bucket sorting for getting information and installing packages?

**If the same application is available in several packages, scoop will use this sorting to determine the "priority" bucket** (now alphabetical sorting is used instead). 

This is useful when using several buckets (I use about 10), and need to install some package whose name I know, but do not want to figure out which bucket it is available in. But I want to install it by priority from certain buckets.
![image](https://user-images.githubusercontent.com/2379473/187659458-e93a6405-b397-4e7f-89a1-e7556a5034ba.png)

Use bucket order for comands:
- info, install, search results, depends, download, cat.

### 1. Current version
```
$> scoop search scoop-completion
Name             Version Source    Binaries
----             ------- ------    --------
scoop-completion 0.2.3   dev-tools
scoop-completion 0.2.3   dodorz
scoop-completion 0.2.3   extras
scoop-completion 0.2.3   naderi

$> scoop install scoop-completion
Installing 'scoop-completion' from dev-tools bucket
```

### 2. Set order
```
Buckets:

1. main
2. extras
3. dodorz
4. naderi
5. dev-tools
```

### 3. Result
```
$> scoop search scoop-completion
Name             Version Source    Binaries
----             ------- ------    --------
scoop-completion 0.2.3   extras
scoop-completion 0.2.3   dodorz
scoop-completion 0.2.3   naderi
scoop-completion 0.2.3   dev-tools

$> scoop install scoop-completion
Installing 'scoop-completion' from extras bucket
```

### Variant 1: Use bucket sorting list
> I assume that this option is easier to implement 

Sorting is used for all buckets, new buckets are added to the end.
We can specify a specific order in the list for each bucket

### Variant 2: Use the 'order' parameter

We can prioritize only the buckets we want - the others will be used alphabetically the way they are prioritized by default.
Each bucket will have an "order" parameter.

**Comment by HUMORCE** [comment]
yes and I think the official bucket should have a higher weighting (by default)

**Comment by chawyehsu** [comment]
```
scoop bucket add 1-main <repo>
scoop bucket add 2-extras <repo>
scoop bucket add 3-foo <repo>
```

This is not recommended. It would break dependencies prefixed with bucket names. e.g.
```
"depends": "extras/vscode"
```

**Comment by rashil2000** [comment]
I feel the best compromise would be to search buckets in the order they are added by user. In this case Main bucket will always be first as it is added by default, and the rest of the bucket order can be left to the user.

**Comment by HUMORCE** [comment]
> It would break dependencies prefixed with bucket names.

just a dirty method, I use it for sort personal bucket first to avoid declaring the bucket name in executing commands only. so, we need this feature.

**Comment by moltenform** [comment]
I'm thinking about the case where the user types `scoop install x`, and there are several buckets that provide a package named x. Instead of getting the first match, maybe we could show a list of potential matches and let the user select which one, before continuing.

**Comment by NamelessUzer** [comment]
> I'm thinking about the case where the user types `scoop install x`, and there are several buckets that provide a package named x. Instead of getting the first match, maybe we could show a list of potential matches and let the user select which one, before continuing.

Users may add buckets other than "main" and "extras", and these buckets may include the same software (the version may or may not be the same).

**Comment by niheaven** [comment]
Yes, will not be closed.

**Comment by God-damnit-all** [comment]
Perhaps the title should be changed to include the word "priority"?
